### ISSUE TITLE
Repeated Downloads Via Browser History

### ISSUE THREAD

**Issue by SnappaGuy** [joshpangell/single-use#33]
Nice but I'm seeing a flaw.

Yes I can make it do everything claimed get a one time download and have the link expire etc, however if I look at the browser history (Chrome in this case) I can find the URL with the admin password ?1234 or whatever its set to and by using that URL generate a new key and do this as many times as I wish.

I can aslo keep refreshing the page and get new keys generated that I can distribute as many times as I wish.

In short, yes each individual link/key is good for one use and a time expiry but I can keep generating new individual links/keys ad infinitum and obtain the file repeatedly.

Advice please?

**Comment by joshpangell** [comment]
The purpose of this project is to be as simple as possible so it can be integrated in whatever way fits the application. It is not meant to be a robust end-to-end too. 

On that note, the generator link is meant to be a simple example that you could extrapolate from. Passing a password via URL query string is not secure in the least. However, doing so, you can easily see how this works so you can call the generator from a secure link, via ajax, curl, or some other method.
